### PR TITLE
ZENKO-2780 basic buckets listing with assumeRoleWithWebIdentity

### PR DESCRIPTION
**Pull request by nicolas2bert**
List accounts with temporary credentials coming from STS.assumeRoleWithWebIdentity call.

Screen Shot 2020-09-10 at 4 33 37 PM

TODO: 
- empty state page
- populate the right layout
- add CreationDate column

### CODE REVIEW

File: public/assets/config.json [hunk 1]
```diff
@@ -2,6 +2,6 @@
     "managementEndpoint": "http://127.0.0.1:5000",
     "oidcAuthority": "http://127.0.0.1:8080/auth/realms/myrealm",
     "oidcClientId": "myclient",
-    "stsEndpoint": "https://sts.amazonaws.com",
-    "s3Endpoint": "https://s3.amazonaws.com"
+    "stsEndpoint": "http://127.0.0.1:8383/sts",
```
Comment (anurag4DSB) [inline public/assets/config.json:5]: Just curious what service is running on this port

File: src/react/actions/sts.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export function assumeRoleWithWebIdentity(role?: string): ThunkStatePromisedAction {
+    return (dispatch, getState) => {
+        const { oidc, auth: { config, stsClient }, configuration } = getState();
+        let roleArn = role;
+        if (!role) {
+            if (configuration.latest.users.length === 0) {
+                return Promise.resolve();
+            }
+            // TODO: which one should we pick?`
```
Comment (anurag4DSB) [inline src/react/actions/sts.js:16]: which one to pick? What do you mean here?

File: src/react/actions/sts.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            roleArn,
+        };
+        return stsClient.assumeRoleWithWebIdentity(assumeRoleParams)
+            .then(creds => {
+                const s3Params = {
+                    endpoint: config.s3Endpoint,
+                    accessKey: creds.Credentials.AccessKeyId,
+                    secretKey: creds.Credentials.SecretAccessKey,
+                    // TODO: to be uncommented once sessionToken is implemented in Vault STS.
+                    // sessionToken: creds.Credentials.SessionToken,
```
Comment (anurag4DSB) [inline src/react/actions/sts.js:31]: Noted 👍 
This one is for me :)

File: src/types/sts.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    +RoleSessionName: string,
+    +roleArn?: string,
+|};
+
+export type AssumeRoleWithWebIdentityResp = {|
+    +Credentials: {|
+        +AccessKeyId: string,
+        +SecretAccessKey: string,
+        // TODO after Vault.assumeRoleWithWebIdentity changes:
+        // SessionToken should not be “optional” value.
```
Comment (anurag4DSB) [inline src/types/sts.js:14]: got it :)

**Review by anurag4DSB — APPROVED** [review]
I tried to review as much as I could. Would definitely need some set of REACT-ing eyes 😆

File: public/assets/config.json (hunk 1 above)
Comment (nicolas2bert) [reply public/assets/config.json:5]: Vault STS service :)

Comment (anurag4DSB) [reply public/assets/config.json:5]: But that's on port 8800

File: src/react/actions/sts.js (hunk 2 above)
Comment (nicolas2bert) [reply src/react/actions/sts.js:16]: For now, if a user owns multiple accounts, by default the application will assume the role of the first account found.
Later, we will probably use `sessionStorage` to keep track of the selected account.

File: public/assets/config.json (hunk 1 above)
Comment (nicolas2bert) [reply public/assets/config.json:5]: There will be a proxy to avoid CORS issues.

**Review by terrydervaux — APPROVED** [review]
LGTM

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**